src/iomusxml.cpp

```diff
@@ -582,6 +582,8 @@ bool MusicXmlInput::ReadMusicXml(pugi::xml_node root)
     // generate page head
     pugi::xpath_node_set credits = root.select_nodes("/score-partwise/credit[@page='1']/credit-words");
     if (!credits.empty()) {
+        bool hasHead = false;
+        bool hasFoot = false;
         PgHead *head = new PgHead();
         PgFoot *foot = new PgFoot();
```

Comment:
This will cause a memory leak if one of them is empty. You don't need to add a `bool` but can simply initialize the pointers with `NULL` and check on that.

Reply:
Fixed in the latest commit. (You may prefer to squash-merge this pull request so the memory leak doesn't make it into the git history).